fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -167,29 +177,201 @@ public Integer selectQueryPartitionId(@Nonnull Tuple groupKey) {
 
     /**
      * return the partition ID on which to run a query, given a grouping key.
-     * the most recent partition is returned, unless the query is sorted by the
-     * partitioning field in ascending order, in which case the oldest partition
-     * is returned.
+     *
+     * If no partitioning field predicate can be used to determine a particular
+     * starting partition, he most recent partition is returned, unless the query
+     * is sorted by the partitioning field in ascending order, in which case the
+     * oldest partition is returned.
+     *
+     * If the query contains a top level partitioning field predicate in conjunction
+     * with the rest of the predicates, and the predicate can be used to determine
+     * a specific starting partition, that partition will be computed and returned.
      *
      * @param groupKey group key
-     * @param sort sort
+     * @param luceneScanQuery query
      * @return partition, or <code>null</code> if partitioning isn't enabled or no partitioning metadata exist
      */
     @Nullable
-    public LucenePartitionInfoProto.LucenePartitionInfo selectQueryPartition(@Nonnull Tuple groupKey, @Nullable Sort sort) {
+    public LucenePartitionInfoProto.LucenePartitionInfo selectQueryPartition(@Nonnull Tuple groupKey, @Nullable LuceneScanQuery luceneScanQuery) {
```

Comment:
I can convert this method as suggested, and then wrap it with another shim that calls it with `asyncToSync`, since ultimately this is called by `LuceneIndexMaintainer.scan()` which is synchronous.